ui/src/store.js

```diff
@@ -0,0 +1,186 @@
+import autodux from 'autodux';
+import { doFetch } from './utils/fetch-websocket';
+
+export const {
+  reducer,
+  initial: defaultState,
+  actions: {
+    activateConnection,
+    deactivateConnection,
+    serverConnected,
+    serverDisconnected,
+    updatePurses,
+    changePurse,
+    swapInputs,
+    changeAmount,
+    createOffer,
+    resetState,
+    updateInvitationDepositId,
+  },
+} = autodux({
+  slice: 'autoswap',
+  initial: {
+    active: false,
+    connected: false,
+    account: null,
+    purses: null,
+    inputPurse: null,
+    outputPurse: null,
+    inputAmount: null,
+    outputAmount: null,
+    freeVariable: null,
+    invitationDepositId: null,
+  },
+  actions: {
+    activateConnection: state => ({ ...state, active: true }),
```

Comment:
Is it better to have this be a particular named method, or to have the payload of a more general method call merely be `{ active: true }` and let autodux deal with it? I don't know the answer.

Reply:
It's a named method for now ⛵ .  I think it's pretty clear.